### PR TITLE
Add an outline to the "Psi: Advanced" window #792

### PR DESCRIPTION
**Pull request by yatigautam**
i have made some changes to the code file, margin adjustments are absolutely fine now 
let me know if any other changes are required.

### CODE REVIEW

File: src/psioptionseditor.cpp [hunk 1]
```diff
@@ -135,11 +135,18 @@ PsiOptionsEditor::PsiOptionsEditor(QWidget *parent) : QWidget(parent)
     tpm_ = new QSortFilterProxyModel(this);
     tpm_->setSourceModel(tm_);
 
-    QVBoxLayout *layout = new QVBoxLayout(this);
-    layout->setContentsMargins(0, 0, 0, 0);
+    QVBoxLayout *mainLayout = new QVBoxLayout(this);
+    mainLayout->setContentsMargins(10, 10, 10, 10); // Adjust margins for the main layout
+
+    QFrame *frame = new QFrame(this);
```
Comment (Ri0n) [inline src/psioptionseditor.cpp:141]: what is this frame for? please attach screenshots

File: src/psioptionseditor.cpp [hunk 2]
```diff
@@ -260,7 +269,7 @@ void PsiOptionsEditor::tv_edit(const QModelIndex &idx)
 #if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
     } else if (value.type() == QVariant::Color) {
 #else
-    } else if (value.typeId() == QMetaType::QColor) {
```
Comment (Ri0n) [inline src/psioptionseditor.cpp:263]: Please make sure the file is formatted with clang-format-18.
You can setup pre-commit.com and also setup proper clang-format version to make sure the file is properly formatted on commit.

File: src/psioptionseditor.cpp [hunk 3]
```diff
@@ -135,11 +135,18 @@ PsiOptionsEditor::PsiOptionsEditor(QWidget *parent) : QWidget(parent)
     tpm_ = new QSortFilterProxyModel(this);
     tpm_->setSourceModel(tm_);
 
-    QVBoxLayout *layout = new QVBoxLayout(this);
-    layout->setContentsMargins(0, 0, 0, 0);
+    QVBoxLayout *mainLayout = new QVBoxLayout(this);
+    mainLayout->setContentsMargins(10, 10, 10, 10); // Adjust margins for the main layout
```
Comment (Ri0n) [inline src/psioptionseditor.cpp:139]: I believe for the detached window you can avoid setting margins at all. 
I guess defaults should be good enough.

**Review by Ri0n — CHANGES_REQUESTED** [review]

**Comment by Ri0n** [comment]
I propose just not setContentsMargins(0, 0, 0, 0); inside the class but set outside just when the widget in the embedded mode. other changes are not needed.

The only thing I would also propose is setSpacing() for the filter line

File: src/psioptionseditor.cpp (hunk 1 above)
Comment (yatigautam) [reply src/psioptionseditor.cpp:141]: This creates a new widget that is a child of the current widget (this). The QFrame can be used to contain other widgets and apply a frame around them, which can be useful for grouping related UI elements together visually.
Screenshot (1)

Comment (yatigautam) [reply src/psioptionseditor.cpp:141]: This creates a new  widget that is a child of the current widget (this). The QFrame can be used to contain other widgets and apply a frame around them, which can be useful for grouping related UI elements together visually.
308119581_497084679136218_7468071683868194394_n

Comment (yatigautam) [reply src/psioptionseditor.cpp:141]: This creates a new widget that is a child of the current widget (this). The QFrame can be used to contain other widgets and apply a frame around them, which can be useful for grouping related UI elements together visually.
Screenshot (246)

**Comment by yatigautam** [comment]
i have understood the changes that are needed, will make the required changes and push them soon.

**Comment by yatigautam** [comment]
hello sir, i wanted to cross-check something, below i have attached some screenshots of the detached window, and it's not looking very well adjusted in minimized as well as maximized mode, i have made some changes and have those screenshots attached above, i think it would look more organized and adjusted if changed, what do you suggest?
Screenshot (248)
Screenshot (247)

**Comment by Ri0n** [comment]
Hi Yati, 
I'm sorry for intervening your work, but I just pushed to master the solution I described above. It looks good to me.
You can checkout and learn from it.

**Comment by yatigautam** [comment]
sure sir.

**Comment by Neustradamus** [comment]
Thanks for your effort @yatigautam!

Original ticket is:
- https://github.com/psi-im/psi/issues/792

Solved by @Ri0n in this commit:
- https://github.com/psi-im/psi/commit/d33895741f930c867cd3331bacf1bbe54e3617bf

**Pull request closed without merging**